proxy/internal/service/init.go

```diff
@@ -49,29 +50,77 @@ func initTrackers(p *Proxy) error {
 }
 
 func initIndex(p *Proxy) error {
-	var err error
-	indexPath := p.config.ModuleStorage.IndexPath
-	p.index, err = store.NewIndex(store.IndexOptions{
-		Directory: indexPath,
-	})
-	return err
+	if p.config.ModuleStorage != nil {
```

Comment:
add check that these are not both nil
add check that these are not both non-nil 
(i.e. mutually exclusive)